lib/StripeResource.js

```diff
@@ -292,6 +292,7 @@ StripeResource.prototype = {
     function makeRequest(apiVersion, headers) {
       var timeout = self._stripe.getApiField('timeout');
       var isInsecureConnection = self._stripe.getApiField('protocol') == 'http';
+      var agent = isInsecureConnection ? self._stripe.getApiField('http_agent') : self._stripe.getApiField('https_agent');
```

Comment:
I assume we have to continue to support `http` for things like dev proxies?

Reply:
That, and I'd like to update the test suite to use stripe-mock instead of local stubbing + real API requests at some point. (Although stripe-mock does support HTTPS, if you disable cert verification.)